module/dkg/broker.go

```diff
@@ -55,28 +56,43 @@ func NewBroker(
 	committee flow.IdentityList,
 	me module.Local,
 	myIndex int,
-	dkgContractClient module.DKGContractClient,
+	dkgContractClients []module.DKGContractClient,
 	tunnel *BrokerTunnel) *Broker {
 
 	b := &Broker{
-		log:               log.With().Str("component", "broker").Str("dkg_instance_id", dkgInstanceID).Logger(),
-		unit:              engine.NewUnit(),
-		dkgInstanceID:     dkgInstanceID,
-		committee:         committee,
-		me:                me,
-		myIndex:           myIndex,
-		dkgContractClient: dkgContractClient,
-		tunnel:            tunnel,
-		privateMsgCh:      make(chan messages.DKGMessage),
-		broadcastMsgCh:    make(chan messages.DKGMessage),
-		shutdownCh:        make(chan struct{}),
+		log:                log.With().Str("component", "broker").Str("dkg_instance_id", dkgInstanceID).Logger(),
+		unit:               engine.NewUnit(),
+		dkgInstanceID:      dkgInstanceID,
+		committee:          committee,
+		me:                 me,
+		myIndex:            myIndex,
+		dkgContractClients: dkgContractClients,
```

Comment:
Writing down some more detail of my suggestion from our call yesterday for sharing the fallback logic between the QCVoter and DKGBroker and keeping the details of the logic abstracted from QCVoter/DKGBroker.

```go
type FallbackStrategy interface {
  // return index of client to use
  ClientIndex() int
  // indicate successful use of client at index
  Success(int)
  // indicate failure to use client at index
  Failure(int)
}

// broker.go
type Broker struct {
  fallbackStrategy FallbackStrategy
  clients []DKGContractClient
}

// ... when using a client
clientIndex := fallbackStrategy.ClientIndex() 
err := clients[clientIndex].CallSomeFunction()
if err != nil {
  fallbackStrategy.Failure(clientIndex)
} else {
  fallbackStrategy.Sucess(clientIndex)
}
```

Reply:
This is probably an (excellent) idea that deserves its own issue.